src/terminal.rs

```diff
@@ -0,0 +1,50 @@
+use crate::Todo;
+use console::Term;
+use std::io::Result;
+
+pub struct Terminal {
+    term: Term,
+}
+impl Terminal {
+    pub fn new() -> Self {
+        Terminal {
+            term: Term::stdout(),
+        }
+    }
+
+    pub fn input(&self) -> Result<String> {
+        self.term.read_line()
+    }
+
+    pub fn ask_for_new_todo(&self) -> Result<Todo> {
+        println!("Qual será seu novo todo? 💬");
+        let todo_name = self.input()?;
+        Ok(Todo::new(todo_name))
+    }
+
+    pub fn ask_if_new_todo(&self) -> Result<bool> {
+        println!("Você deseja criar um novo TODO? ⛏  (insira apenas n ou s)");
+        let response: String = self.input()?;
```

Comment:
O compilador já vai conseguir inferir o tipo certo.

```suggestion
        let response = self.input()?;
```

Reply:
Feito